NYPL/fedora_ingest_rails
# DR-2206 remove hierarchicalgeographic mtxt from repapi docs

## Jira Tickets ##
[unknown field 'hierarchicalgeographic_mtxt' in repoapi solr](https://jira.nypl.org/browse/DR-2206)

## Things Done ##
- Added this field to the method that removes unwanted fields from repoapi docs
- Added test coverage

## How to Test ##
- Set `REPO_SOLR_URL=http://10.225.131.109:8983/solr/repoapi` in your .env to point to the dev repo SOLR with the repoapi index
- Go through the following in the rails console on the `develop` branch:
```
repo_solr = RepoSolrClient.new
uuid = '4b7b9750-7f28-0133-8e0f-00505686d14e'
mms_client = MMSClient.new(mms_url: "http://qa-metadata.nypl.org", user_name: 'admin', password: 'password')
parent_and_item_repo_docs = mms_client.repo_docs_for(uuid)
repo_solr.add_docs_to_solr(parent_and_item_repo_docs)
```
You should receive an error on the last command because of the unwanted field `hierarchicalgeographic mtxt`
- Check out the PR branch and do the same. The doc should be added to SOLR without error.
- Run the rspecs
